File: lib/python/src/wg_portal_client/metrics.py
from ._http import _HttpClient
from ._models import InterfaceMetrics, PeerMetrics, UserMetrics


class MetricsApi:
    """API methods for WireGuard traffic metrics."""

    def __init__(self, http: _HttpClient) -> None:
        self._http = http

    def by_interface(self, interface_id: str) -> InterfaceMetrics:
        return self._http.get(f"/metrics/by-interface/{interface_id}")  # type: ignore[return-value]

    def by_peer(self, peer_id: str) -> PeerMetrics:
        return self._http.get(f"/metrics/by-peer/{peer_id}")  # type: ignore[return-value]

    def by_user(self, user_id: str) -> UserMetrics:
        return self._http.get(f"/metrics/by-user/{user_id}")  # type: ignore[return-value]

File: lib/rust/src/metrics.rs
use crate::http::{HttpClient, WgPortalError};
use crate::models::{InterfaceMetrics, PeerMetrics, UserMetrics};

pub struct MetricsApi<'a> {
    http: &'a HttpClient,
}

impl<'a> MetricsApi<'a> {
    pub(crate) fn new(http: &'a HttpClient) -> Self {
        Self { http }
    }

    pub fn by_interface(&self, interface_id: &str) -> Result<InterfaceMetrics, WgPortalError> {
        self.http
            .get(&format!("/metrics/by-interface/{}", interface_id), None)
    }

    pub fn by_peer(&self, peer_id: &str) -> Result<PeerMetrics, WgPortalError> {
        self.http
            .get(&format!("/metrics/by-peer/{}", peer_id), None)
    }

    pub fn by_user(&self, user_id: &str) -> Result<UserMetrics, WgPortalError> {
        self.http
            .get(&format!("/metrics/by-user/{}", user_id), None)
    }
}
